### ISSUE TITLE
Re-investigate focus states with invalid form items

### ISSUE THREAD

**Issue by tw15egan** [carbon-design-system/carbon#6529]
Based on https://github.com/carbon-design-system/carbon/issues/6513
Related to https://github.com/carbon-design-system/carbon/pull/6522

## What package(s) are you using?

- [x] `carbon-components`

## Detailed description

> Describe in detail the issue you're having.

When an item is invalid, the invalid border takes precedence over the focus state. This causes a11y concerns, as a keyboard user has no visual way to navigate through multiple invalid form items. 

> Is this issue related to a specific component?

Form items (e.g. Text Input, Textarea, DatePicker)

> What did you expect to happen? What happened instead? What would you like to
> see changed?

I would expect the form item to receive a focus border when focused, even when invalid

> What browser are you working in?

Chrome

> What version of the Carbon Design System are you using?

Latest

## Steps to reproduce the issue

1. Go to https://react.carbondesignsystem.com/?path=/story/textinput--default
2. Go to knobs --> set the invalid state to true
3. With an invalid input, click into the text input and notice no focus state is added
4. Tab through and notice no visual change on focus

## Additional information

![form](https://user-images.githubusercontent.com/11928039/88311213-77473380-ccc5-11ea-8f54-51f5ebcb40d4.gif)

**Comment by laurenmrice** [comment]
This spec includes invalid states when they are closed, open with focus present and with an element that is focused in the menu.

Spec:
![Artboard](https://user-images.githubusercontent.com/43969356/88324469-5c86b600-cce9-11ea-9ed0-8c4ed5420b04.png)
